GSA_Adapter/CRUD/Read.cs

```diff
@@ -200,7 +203,7 @@ public List<Node> ReadNodes(List<string> ids = null)
         public List<ISectionProperty> ReadSectionProperties(List<string> ids = null)
         {
             List<IMaterialFragment> matList = ReadMaterials(null, true);
-            Dictionary<string, IMaterialFragment> materials = matList.ToDictionary(x => x.CustomData[AdapterIdName].ToString());
+            Dictionary<string, IMaterialFragment> materials = matList.ToDictionary(x => (x.GSAId().ToString() ?? x.Name).ToString());
```

Comment:
With the Id now being obejct, can store default names with string in the Id
```suggestion
            Dictionary<string, IMaterialFragment> materials = matList.ToDictionary(x => GetAdapterId(x).ToString());
```